packages/interface-mocks/src/muxer.ts

```diff
@@ -399,6 +394,19 @@ class MockMuxer implements StreamMuxer {
 
     return muxedStream
   }
+
+  close (err?: Error): void {
+    if (this.closeController.signal.aborted) return
+    this.log('closing muxed streams')
+
+    if (err == null) {
+      this.streams.forEach(s => s.close())
+    } else {
+      this.streams.forEach(s => s.abort(err))
+    }
+    this.closeController.abort()
+    this.input.end(err)
```

Comment:
Calling `this.input.end(err)` may not be necessary here.

I think what will happen is calling `this.closeController.abort()` will cause the `pipe` in `sink` to throw, the error will be caught and passed to `this.input.end` on line 316.

Ending the input should then cause the connection to close?  It's not clear what use a muxed connection with a closed muxer is, so that's probably good.

Reply:
Right, I think the only big difference is that by calling `this.input.end(err)` explicitly, the error is passed to the source.
If we rely on the pipe in sink to throw, the error that the source will get will be some generic "aborted" error.

Not sure which is the desired behavior, or that either behavior matters to the interface.

Reply:
Maybe `this.input.end(err)` is better - it's not async and the received error is more useful than 'aborted'.